lib/ansible/utils/display.py

```diff
@@ -484,7 +502,14 @@ def get_deprecation_message(self, msg, version=None, removed=False, date=None, c
         return message_text
 
     @proxy_display
-    def deprecated(self, msg, version=None, removed=False, date=None, collection_name=None):
+    def deprecated(
+        self,
+        msg: str,
+        version: t.Any = None,
```

Comment:
We use either `str` of `float` but anything should do since we just run the value through `str.format()`. Same for `date` below.

Reply:
I would use `object` instead of `Any` here. `Any` allows accessing any arbitrary member of the object (i.e. it effectively disables type checking for that variable) while with `object`, the type checker will only allow the common denominator (e.g. `.__str__()`).

Reply:
Sounds like
```suggestion
        version: float | str | None = None,
```
to me.

Reply:
@mkrizek is there any actual use of `float` in the code base? GH search didn't reveal anything. Maybe, just keep `str` and fix any violations?

Reply:
Unless you want to document that the API coerces any object into a string, then yeah, it makes more sense to use `str | None`. If you *do* want that for whatever reason, then `object | None` is better than `Any` in my opinion.

Reply:
@mkrizek any opinion here?

Reply:
I fixed the `float` occurrences to be strings and annotated that the `version` is a string since that should cover all version formats. But if people think `str | float | None` is better to align with the existing code without doing any changes to it, I am fine with that too.

Reply:
@mkrizek so you fixed the invocations but not the typing on this line? Looks like this still needs to be done, then:
```suggestion
        version: str | None = None,
```

Reply:
Ah, good catch. I only fixed `get_deprecation_message` and this one I missed.